blogs/2025/11/03/universe-recap.md

```diff
@@ -0,0 +1,118 @@
+
+With all these choices, things got better - and a little more fragmented. Subscription hopping, tool juggling, and FOMO became the norm. This year at GitHub Universe, we set out to fix that with a unified agent experience in VS Code. Now you can switch between agents - local or cloud - and manage them all from one place.
+
+### OpenAI Codex Integration
+
+[OpenAI Codex](https://openai.com/codex/) was a hit with developers in 2025. Riding the wave of GPT-5 and Codex, OpenAI delivered a top-tier agent experience with the Codex CLI and a solid VS Code extension that brings Codex right into the editor.
+
+This year at GitHub Universe, we announced you can use OpenAI Codex with your GitHub Copilot Pro+ login.
+
+To use this integration, install the OpenAI Codex extension and sign in with your Copilot Pro+ account.
+
+![OpenAI Codex sign-in panel in VS Code](codex-signin.gif)
+
+When you use Codex with Copilot Pro+, Copilot handles all model calls and standard rate limits apply. You get code generation, code explanation, and all the features - no need to manage a separate OpenAI account.
+
+With Codex, you now have four powerful coding agents in VS Code:
+
+1. GitHub Copilot
+2. GitHub Copilot Coding Agent (Cloud)
+3. GitHub Copilot CLI
+4. OpenAI Codex
+
+But with all these agents, it's easy to get overwhelmed. What agents are running? Where are they running? What day is it?
+
+That's why we've introduced a new feature in VS Code for orchestrating all your agents - local or remote. We call it, "Agent Sessions".
+
+### Agent Sessions
+
+There's a new view in the VS Code sidebar called "[Agent Sessions](https://code.visualstudio.com/updates/v1_105#_agent-sessions)". It gives you one place to manage all your agents - local or cloud.
+
+![VS Code window with Agent Sessions sidebar showing Copilot, Coding Agent, CLI, and Codex statuses against a calm gray workspace](agent-sessions.png)
+
+With Agent Sessions, you see all agent sessions for your project. You can check which agents are running, their status, and jump between sessions with a click.
+
+Cloud Agent sessions now have a tabbed experience that lets you course-correct the agent mid-run. It's common to send a prompt and realize you forgot something important. Before, you had to wait or cancel. Now, just open the tab, add an update, and watch the agent adjust its plan.
+
+You can also delegate any task to any agent right from the chat.
+
+![VS Code showing the "Delegate" button from the chat, when clicked opens a menu of agents to delegate to](why-settle.gif)
+
+This unified Agent Sessions view makes VS Code a "mission control" for orchestrating all your agents, while keeping you in the editor where you do your best work.
```

Comment:
We should also mention that more agents are on the way.. i think there was a screenshot from GitHub Universe... maybe pibogg can telll us what we can say.